lib/generators/pies.rb

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env ruby
+
+# This test generates a pseudo-random sequence by multiplying pi
+# by an incrementing counter, dividing by e and sin of the product
+
+require_relative '../generator.rb'
+
+class PiEinTheSin < Generator
+
+  def pimulti(i)
+    i * Math::PI
+  end
+
+  def divE(num)
+    num / Math::E
+  end
+
+  def initialize(*args)
+    super(*args)
+    @i = 0
+  end
+
+  def self.summary
+    "Generates a PRN using algo Sin((Pi * i) / e )"
+  end
+
+  def self.description
+    desc = <<-DESC_END
+    This generatior produces a pseudo-random sequence by Sin((Pi * i) / e )
+    DESC_END
+    desc.gsub(/\s+/, " ").strip
+  end
+
+
+  def next_chunk
+    @i += 1
+    Math::sin(divE(pimulti(@i))).to_s

```

Comment:
The goal of my suggestion is not to improve performance. Rather, it is to modify the generator to output a broader range of byte values.

The generator as it is currently implemented creates human-readable [ASCII](https://en.wikipedia.org/wiki/ASCII) strings of the floating-point result of the `sin` operation. So, each byte is limited to the following values: `0-9`, `.`, and `-`. You can see this by running the following:

```
./bin/entrogen pieinthesin -l 128
0.91508850748424030.7380244659037377-0.3198672717046207-0.995999372624937-0.48341230071929160.60612440191802230.9722553299324798
```

This limited range of values is more clear when the byte values are inspected with `hexdump`:
```
./bin/entrogen pieinthesin -l 128 | hexdump 
0000000 2e30 3139 3035 3838 3035 3437 3438 3432
0000010 3330 2e30 3337 3038 3432 3634 3935 3330
0000020 3337 3737 302d 332e 3931 3638 3237 3137
0000030 3037 3634 3032 2d37 2e30 3939 3935 3939
0000040 3733 3632 3432 3339 2d37 2e30 3834 3433
0000050 3231 3033 3730 3931 3932 3631 2e30 3036
0000060 3136 3432 3034 3931 3831 3230 3332 2e30
0000070 3739 3232 3535 3233 3939 3233 3734 3839
```

Note that the byte values are limited to `0x30`-`0x39` (the values for the ASCII characters 0-9), `0x2e` (the ASCII character `.`), and `0x2d` (the ASCII character `-`).

The intent of my suggestion was to read the underlying byte values of the [floating point representation](https://en.wikipedia.org/wiki/Double-precision_floating-point_format#IEEE_754_double-precision_binary_floating-point_format:_binary64) of the `sin` evaluation. The [pack](http://ruby-doc.org/core-2.2.0/Array.html#method-i-pack) method reads the underlying values into a string representation.

Compare the above to the output after the modification:
```
./bin/entrogen pieinthesin -l 128
�xԿW�:���ϳ�%:�޿ttq_e�?}~�5��?��:����?e��ڦ���%ΖMO뿛�ߜ���?�������?J�lb5�?��d~ݿY܅��￦�Y��kֿ
```

```
./bin/entrogen pieinthesin -l 128 | hexdump 
0000000 e370 b192 4867 3fed 8a9d 7c16 9de5 3fe7
0000010 0dcc 93c2 78b4 bfd4 5704 1388 df3a bfef
0000020 b3cf 2584 f03a bfde 7474 0071 655f 3fe3
0000030 7e7d 35ad 1cb7 3fef de90 fc3a c8e1 3fc6
0000040 a765 daed 84a6 bfea ce25 4d96 154f bfeb
0000050 829b 9cdf b394 3fc2 84e1 edd7 da9c 3fee
0000060 cb4a 6c19 3562 3fe4 18b6 64e0 1c7e bfdd
0000070 dc59 1385 f2a0 bfef b1a6 8c59 6bba bfd6
```

You will see that there is much more variety in the byte output, and the generator creates much more interesting output when run through `entrospect`.